JS/module/Auth/conditions.js

```diff
@@ -0,0 +1,60 @@
+import { USER_INPUT_ID, requestSettings } from "./requestSettings.js";
+
+// 빈 칸인지 검사하는 함수
+export function checkFormEmpty(inputTag) {
+  const isNotEmpty = !(inputTag.length === 0);
+  return isNotEmpty ? true : false;
+}
+
+//올바른 이메일 형식인지 검사하는 함수
+export function checkEmailValid(email) {
+  const EMAIL_REGEX = new RegExp(
+    /^[0-9a-zA-Z]([-_\.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_\.]?[0-9a-zA-Z])*\.[a-zA-Z]{2,3}$/i
+  );
+  const isValid = EMAIL_REGEX.test(email.trim());
+  return isValid ? true : false;
+}
+
+//올바른 패스워드 형식인지 검사하는 함수
+export function checkPasswordValid(password) {
+  const PASSWORD_REGEX = new RegExp(
+    /^(?=.*\d)(?=.*[a-zA-Z])[0-9a-zA-Z]{8,16}$/
+  );
+  const isValid = PASSWORD_REGEX.test(password.trim());
+  return isValid ? true : false;
+}
+
+//두 값을 비교해서 일치하는지 검사하는 함수
+export function checkStringSame(value1, value2) {
+  return value1 === value2 ? true : false;
+}
+
+//사용 가능한 이메일인지 검사하는 함수
+export async function checkEmailAvailable() {
+  try {
+    const ID = USER_INPUT_ID();
+    const response = await fetch(
+      "https://bootcamp-api.codeit.kr/api/check-email",
+      requestSettings("POST", ID)
+    );
+    if (response.status === 200) {
+      return true;
+    } else if (response.status === 409) {
+      return false;
```

Comment:
경서님과 멘토님이 말씀해주신 얼리 리턴 패턴 저도 자주 사용하는 방식인데요! (코테할 때도 많이 썼었죵,,ㅎㅎ 코테할 땐 그냥 바로 함수를 끝내버리니까 쓸데없는 계산 더 안해도 되고 런타임 절약할 때 아주 꿀템입니다.)
```suggestion
    if (response.status === 200) return true;
    return false;
```
저라면 이렇게 작성할 것 같아요!
참고로 개인 취향이지만 if문 내에 코드가 한 줄 뿐일 때 전 중괄호는 생략하는 편입니다,, 안그래도 긴 파일인데 중괄호가 괜히 2줄이나 까먹는게 너무 시러요.... 그리고 특히 이 return문은 누구나 이해할 수 있는 코드다 보니! 저라면 1줄로 적을 것 같습니당 😖

얼리 리턴 패턴을 사용하면 200인 애들은 이미 return이 되어서 함수 밖으로 나간 거니까 다음 줄부터는 200인 애들은 아예 고려안해도 된다는 장점! 이 있습니당
어쩌다보니 얼리 리턴 패턴 홍보대사....(?)